ui/packages/catalog-ui-search/src/main/webapp/react-component/container/sharing/sharing.tsx

```diff
@@ -113,30 +48,36 @@ export class Sharing extends React.Component<Props, State> {
       items: [],
     }
   }
-
   componentDidMount = () => {
     fetch(`/search/catalog/internal/metacard/${this.props.id}`)
       .then(response => response.json())
       .then(data => {
         const metacard = data.metacards[0]
-        const items = getGroups(
-          metacard[Security.GroupsWrite] || [],
-          metacard[Security.GroupsRead] || []
-        ).concat(
-          getIndividuals(
-            metacard['metacard.owner'] || [],
-            metacard[Security.IndividualsWrite] || [],
-            metacard[Security.IndividualsRead] || [],
-            metacard[Security.AccessAdministrators] || []
-          )
-        )
-        this.setState({ items: items })
+        const res = Restrictions.from(metacard)
+        const security = new Security(res)
+        const individuals = security.getIndividuals().map((e: Entry) => {
```

Comment:
✏️ Since both lambdas produce essential the same object (one using group and the other user for category) I would like to see it pulled up into a single entity that can be re-used between the two